### ISSUE TITLE
Potential issue with Exodus plugin in obtaining element block count

### ISSUE THREAD

**Issue by aowen87** [visit-dav/issues-test#2638]
So, I've seen this in some data Rick Angelini gave me.

Its 32-way Exodus data. Some files define 3 element blocks, some 4 and some 5.

I've opened this data in serial VisIt, the *first* file it examines has 3 element blocks and so it decides there are only 3 element blocks during populate database metadata. The additional blocks in those files that have 'em are ignored. Thats BAD. Simply changing which file VisIt thinks is the *first* file to open so that it opens a file with 5 element blocks _fixes_ this.

In parallel, I seriously have no idea how VisIt can get it right because there appears to be no communication regarding the element block count. Thats bad.

OTOH, we don't want VisIt opening *all* files in an ensemble just to obtain element block count.

What is the solution?
* Require that all Exodus data producers always indicate global element block count in each file. For non-existent blocks, the exodus file will have to have a zero element block for those blocks. That basically fixes the problem by enforcing data producer to produce uniform exodus data. 
* Provide database open option to specify the element block count manually?
* Provide !NBLOCKS-like directive, !NELEM_BLOCKS that indicates the block count
* Just require that any .visit/.exodus file have the file with greatest element block count at the head of the list
* Open all the files and find the global element block count. In parallel this may not be as bad as it sounds.

Note that this is identically the same problem we have with Mili and for which we had the unsatisfactory solution of a makemili tool. The long term solution we are hoping to get LLNL Engineering customers to use there is for data producer to do effectively the work makemili would have done but at mc_close (Mili close) time. By adding logic to Mili's database close method *and* requiring data producer call it collectively on all processors, we can do work in Mili library to gather up information from individual ranks and then write the unified information to the mili database.

It might be worth thinking about a generalized strategy for this for products like Mili, Exodus, netCDF, VTK, Silo, that would be available to codes to just sort of use out the box.
 

-----------------------REDMINE MIGRATION-----------------------
This ticket was migrated from Redmine. The following information
could not be accurately captured in the new ticket:

Original author: Mark Miller
Original creation: 02/07/2017 08:59 pm
Original update: 02/14/2017 06:36 pm
Ticket number: 2755
